### PR TITLE
[REF-5699] Enable user's metrics definition with REST model serving

### PR DESCRIPTION
**Pull request by zohar-mizrahi**

### CODE REVIEW

File: reflex-algos/src/main/python/mlcomp/parallelm/common/topological_sort.py [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+        def childs(self):
+            return self._childs
+
+        def __str__(self):
+            child_keys = [c.key for c in self.childs] if self.childs else None
+            return "key: {}, childs: {}".format(self.key, child_keys)
+
+    logging.basicConfig(level=logging.INFO, format='%(name)s %(levelname)s: %(message)s')
+
+    n1 = Node("a", None)
```
Comment (liororama) [inline reflex-algos/src/main/python/mlcomp/parallelm/common/topological_sort.py]: Can you add this tests as a unit tests.. mlcomp has unite tests

File: reflex-algos/src/main/python/mlcomp/parallelm/components/restful/metric.py [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+
+    _metrics = {}
+
+    def __init__(self, name, title=None, hidden=False, metric_type=MetricType.COUNTER, value_type=int,
+                 metric_relation=None, related_metric=None):
+        super(Metric, self).__init__(logging.getLogger(self.logger_name()))
+
+        self._metric_name = name + Metric.NAME_SUFFIX
+        self._title = title
+        self._hidden=hidden
```
Comment (liororama) [inline reflex-algos/src/main/python/mlcomp/parallelm/components/restful/metric.py]: space before and after =

**Review by liororama — APPROVED** [review]

File: reflex-algos/src/main/python/mlcomp/parallelm/common/topological_sort.py (hunk 1 above)
Comment (zohar-mizrahi) [reply reflex-algos/src/main/python/mlcomp/parallelm/common/topological_sort.py]: It is already there. I left this code here to enable me testing it as a self contained python file.

File: reflex-algos/src/main/python/mlcomp/parallelm/components/restful/metric.py [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+        """
+        if uwsgi_loaded:
+            if self._value_type == float:
+                delta *= Metric.FLOAT_PRECISION
+
+            uwsgi.metric_inc(self._metric_name, int(delta))
+
+    def decrease(self, delta=1):
+        """
+        increase the metric's value by the given delta
```
Comment (yakov-g) [inline reflex-algos/src/main/python/mlcomp/parallelm/components/restful/metric.py]: decrease in the comment

**Review by yakov-g — APPROVED** [review]

**Pull request closed without merging**